src/client/components/dashboard/robot_panel/view_model.ts

```diff
@@ -0,0 +1,75 @@
+import { computed } from 'mobx'
+import { createTransformer } from 'mobx'
+import { message } from '../../../../shared/proto/messages'
+import { DashboardRobotModel } from '../dashboard_robot/model'
+import { LastStatus } from './view'
+import State = message.behaviour.Behaviour.State
+
+import Mode = message.input.GameState.Data.Mode
+import PenaltyReason = message.input.GameState.Data.PenaltyReason
+import Phase = message.input.GameState.Data.Phase
+
+export class RobotPanelViewModel {
+  public constructor(private model: DashboardRobotModel) {
+  }
+
+  public static of = createTransformer((model: DashboardRobotModel) => {
+    return new RobotPanelViewModel(model)
+  })
+
+  @computed
+  public get batteryValue(): string {
+    const battery = this.model.battery
+    return battery === -1 ? '' : `${Math.round(battery * 100)}%`
+  }
+
+  @computed
+  public get behaviour(): string {
+    return State[this.model.behaviourState] || State[State.UNKNOWN]
+  }
+
+  @computed
+  public get lastCameraImage(): LastStatus {
+    return this.getLastStatus(this.model.lastCameraImage, 3)
+  }
+
+  @computed
+  public get lastSeenBall(): LastStatus {
+    return this.getLastStatus(this.model.lastSeenBall, 5)
```

Comment:
Can you make it a configuration value so we can tweak it? It'll be hard to know until an actual game. It's probably more related to the covariance of the ball/field than a time.